explorer/syntax/parser.ypp

```diff
@@ -108,6 +108,7 @@
 %type <ClassExtensibility> class_declaration_extensibility
 %type <std::optional<Nonnull<Expression*>>> class_declaration_extends
 %type <Nonnull<Declaration*>> declaration
+%type <DeclaredName> declared_name
```

Comment:
In the implementation you say you need a default constructor for this, but couldn't use also switch DeclaredName to being a pointer and Nonnull it as we do elsewhere, or alternately make it a std::optional? I do wonder whether supporting default construction so that you can write it this way is best...

Reply:
Especially given the focus on value types throughout the Carbon design and the intent for the Carbon implementation to use the same ideals as in the Carbon language design to the extent that we can, it seems to me that we should also use value types where possible in explorer. I don't think the fact that bison forces default constructibility on us is good enough reason to introduce a non-value type.

Reply:
```suggestion
%type <BisonWrap<DeclaredName>> declared_name
```

Your comments reminded me, I added a wrapper for this. :)